### PR TITLE
Add WebNN implementation status contribution mechanics

### LINKED ISSUE #66: Lack of WebNN implementation status contribution mechanics
There is no contribution mechanics to update [WebNN implementation status](https://webmachinelearning.github.io/webnn-status/).

### PR DESCRIPTION
**Pull request by ibelem**
Fix https://github.com/webmachinelearning/webmachinelearning.github.io/issues/66 .

@anssiko PTAL

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Contribution
```
Comment (anssiko) [inline CONTRIBUTING.md]:
```suggestion
# Contributing to Web Machine Learning website
```

Good to make it explicit the scope here is the website to differentiate it from e.g. the WebNN API specification.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+First of all, thank you very much for your interest in contributing to Web Machine Learning website! We're really happy to accept contributions from everyone, whether you're a browser engineer, a spec writer or a web developer.
+
+## WebNN Implementation Status
+
+This repository contains raw data from the [WebNN Implementation Status](https://webmachinelearning.github.io/webnn-status/) tables. If you are interested in updating or adding to the WebNN Implementation Status data on the site, please edit the [webnn_status.json](./assets/json/webnn_status.json) file.
+
+The WebNN API [browser compat data (BCD)](https://github.com/mdn/browser-compat-data/pull/22569) is the upstream and we will keep it up to date as a first priority. The [webnn_status.json](./assets/json/webnn_status.json) raw data is the enhancement of [WebNN API BCD](https://github.com/mdn/browser-compat-data/pull/22569).
```
Comment (anssiko) [inline CONTRIBUTING.md]:
```suggestion
The WebNN API [browser compat data (BCD)](https://github.com/search?q=repo%3Amdn%2Fbrowser-compat-data+webnn) is the upstream and we will keep it up to date as a first priority. The [webnn_status.json](./assets/json/webnn_status.json) adds the following domain-specific data on top of the [WebNN API BCD](https://github.com/search?q=repo%3Amdn%2Fbrowser-compat-data+webnn) baseline data:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+- WebNN BCD data is unable to support WebNN mapped ops of each native backends
+- WebNN BCD data doesn't include WPT tests and test results
+- WebNN BCD data is unable to cover JavaScript ML Frameworks integration status
```
Comment (anssiko) [inline CONTRIBUTING.md]:
```suggestion
- WebNN ops mapping to native backends
- WebNN WPT test and test result links
- WebNN JavaScript ML Frameworks integration status
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+### WebNN Implementation Status Data
+
+The [webnn_status.json](./assets/json/webnn_status.json) includes every feature found for [WebNN Implementation Status](https://webmachinelearning.github.io/webnn-status/). Maintaining this file on GitHub allows anyone to update or contribute to the support data on the site.
```
Comment (anssiko) [inline CONTRIBUTING.md]:
```suggestion
The [webnn_status.json](./assets/json/webnn_status.json) is the data source for the [WebNN Implementation Status](https://webmachinelearning.github.io/webnn-status/) page. By contributing updates to this file on GitHub you can help keep the implementation status page up to date. An example entry for the [`argMax`](https://www.w3.org/TR/webnn/#dom-mlgraphbuilder-argmax) op looks like the following:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      ],
+      "ort_progress": 4,
+      "ort_version_added": "main",
+      "notes": ""
+    }
+```    
+
+### Supported Changes
+
+Currently the following feature information can be modified for WebNN XNNPACK, DirectML and MLService backends.
```
Comment (anssiko) [inline CONTRIBUTING.md]:
```suggestion
Currently the following feature information can be modified for WebNN XNNPACK, DirectML and MLService backends:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+* **op** - Operation name defined in [Web Neural Network API](https://www.w3.org/TR/webnn/) spec
+* **op_id** - Operation name defined in anchor link in the spec
+* **wpt** - The WebNN test cases id used in [Web Platform Tests](https://github.com/web-platform-tests/wpt/tree/master/webnn) and [WPT dashboard](https://wpt.fyi/)
+* **[*]_op** - The operation list of native ML libraries or the JavaScript ML frameworks that map to WebNN API
+* **[*]_version_added** - The Chromium version or the JavaScript ML frameworks started to support this feature
```
Comment (anssiko) [inline CONTRIBUTING.md]:
```suggestion
* **[*]_version_added** - The Chromium or the JavaScript ML framework version where support for the feature was added
```

File: README.md [hunk 7]
```diff
@@ -7,8 +9,18 @@ gem install bundler jekyll
...
 
 ## Run
+
 ```
 bundle install
 bundle exec jekyll serve
 Server address: http://127.0.0.1:4000/
 ```
+
+## Contributions
```
Comment (anssiko) [inline README.md]:
```suggestion
## Contributing
```

File: README.md [hunk 8]
```diff
@@ -7,8 +9,18 @@ gem install bundler jekyll
...
+
+We welcome contributions! Please see the [contribution guidelines](CONTRIBUTING.md).
```
Comment (anssiko) [inline README.md]:
```suggestion
We welcome contributions! Please see the [contributing guidelines](CONTRIBUTING.md).
```

**Review by anssiko — APPROVED** [review]
Thanks for this initial contribution! This looks great overall, added some suggestions.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Ah, you helped to find the best URL to query WebNN BCD files, thanks! Fixed.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 6 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Fixed

File: README.md (hunk 8 above)
Comment (ibelem) [reply README.md]: Fixed

File: README.md (hunk 7 above)
Comment (ibelem) [reply README.md]: Fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ibelem) [reply CONTRIBUTING.md]: Fixed.

**Comment by ibelem** [comment]
@anssiko Luckily I am still in the office and your feedback has been addressed and updated in the new commit, PTAL

**Comment by anssiko** [comment]
Thanks you the updates! Happy to merge this now.

To surface these guidelines more prominently, we can add a direct link at the bottom of https://webmachinelearning.github.io/webnn-status/ e.g.:

`Interested in contributing to this implementation status page? See <a href="...">contributing guidelines</a>.`

**Pull request merged**